### ISSUE TITLE
Support autoregressive decoding of LLMs

### ISSUE THREAD

**Issue by tengyifei** [AI-Hypercomputer/torchprime#183]
For LLMs, we need to support running the model autoregressively to decode out a stream of tokens given some starting token.

This will be useful for quantitatively evaluating the capability of a trained model.

Note that "eval" in this case means inference, and not simply evaluating the cross-entropy loss.
